client/scripts/views/components/new_thread_form.ts

```diff
@@ -134,42 +143,120 @@ export const NewThreadForm: m.Component<{
 }, {
   activeTag: OffchainTag | string,
   autoTitleOverride,
-  error,
   form: IThreadForm,
   fromDraft?: number,
   newType: string,
   quillEditorState,
+  recentlySaved: number[],
   saving: boolean,
   uploadsInProgress: number,
 }> = {
-  view: (vnode) => {
+  oninit: (vnode_) => {
+    console.log('initing');
+    console.log(vnode_.state.form);
+    vnode_.state.form = {};
+    vnode_.state.recentlySaved = [];
+    vnode_.state.uploadsInProgress = 0;
+    if (vnode_.state.newType === undefined) {
+      vnode_.state.newType = localStorage.getItem(`${app.activeId()}-post-type`) || PostType.Discussion;
+    }
+    if (vnode_.state.newType === PostType.Discussion) {
+      vnode_.state.form.threadTitle = localStorage.getItem(`${app.activeId()}-new-discussion-storedTitle`);
+    } else {
+      vnode_.state.form.url = localStorage.getItem(`${app.activeId()}-new-link-storedLink`);
+      vnode_.state.form.linkTitle = localStorage.getItem(`${app.activeId()}-new-link-storedTitle`);
+    }
+  },
+  view: (vnode: m.VnodeDOM<{
+    header: boolean,
+    isModal: boolean
+  }, {
+    activeTag: OffchainTag | string,
+    autoTitleOverride,
+    form: IThreadForm,
+    fromDraft?: number,
+    newType: string,
+    quillEditorState,
+    recentlySaved: number[],
+    saving: boolean,
+    uploadsInProgress: number,
+  }>) => {
```

Comment:
We should avoid using VnodeDOM here because it's not really supposed to be supported by Mithril at the component level. Instead we should use e.target (and look for the target's parent/$(target).closest())

Reply:
Fixed